tensorflow_datasets/core/features/image_feature_test.py

```diff
@@ -109,6 +109,22 @@ def test_image_shaped(self):
         ],
     )
 
+  def  test_image_dtype(self):
+      img = randint(256 , size=(128,128,3) ,dtype=np.uint8)
+      
+      # Image with statically defined dtype   
+      self.assertFeature(
+        feature=features_lib.Image( dtype=tf.uint16),
+        shape=(None, None, 3),
+        dtype=tf.uint8,
+        tests=[
+            testing.FeatureExpectationItem(
+                value = img , 
+                raise_cls = ValueError , 
+                raise_msg = 'dtype should be'
+            ),
```

Comment:
It is good to test the error case, but you should also add a success case where the image is correctly decoded.

Reply:
Thanks for the review @Conchylicultor  , 
But when I run a new test case as following :
```
img_file_path = os.path.join(os.path.dirname(__file__),
                                 '../../testing/test_data/6pixels.png')
img_file_expected_content = [  # see tests_data/README.md
        [[0, 255, 0], [255, 0, 0], [255, 0, 255]],
        [[0, 0, 255], [255, 255, 0], [126, 127, 128]],
    ]
self.assertFeature(
        feature=features_lib.Image(dtype=tf.uint16),
        shape=(None, None, 3),
        dtype=tf.uint16,
        tests=[
            testing.FeatureExpectationItem(
                value = img_file_path , 
                expected = img_file_expected_content
            ),
        ],
    )
```
I was encountered with this error , couldn't figure out where did I go wrong .
```
AssertionError: 
Arrays are not equal

not equal where = (array([0, 0, 0, 0, 1, 1, 1, 1, 1, 1]), array([0, 1, 2, 2, 0, 1, 1, 2, 2, 2]), array([1, 0, 0, 2, 2, 0, 1, 0, 1, 2]))
not equal lhs = [65535 65535 65535 65535 65535 65535 65535 32382 32639 32896]
not equal rhs = [255 255 255 255 255 255 255 126 127 128]
Mismatch: 55.6%
Max absolute difference: 65280
Max relative difference: nan
 x: array([[[    0, 65535,     0],
        [65535,     0,     0],
        [65535,     0, 65535]],...
 y: array([[[  0, 255,   0],
        [255,   0,   0],
        [255,   0, 255]],...

```
Please help me!!

Reply:
That is what tests are for.

It seems that the image that you're trying to decode is encoded as int8 and you're trying to decode it as int16, so the decoding do not provide the correct result. You should instead create a new png image encoded as int16.

Additionally, you should also test when passing a np.ones(dtype=np.int16) directly that it is decoded correctly.